src/super_gradients/training/models/segmentation_models/segformer.py

```diff
@@ -372,12 +386,58 @@ def _remove_auxiliary_heads(self):
     def replace_head(self, new_num_classes: int, new_decoder_embed_dim: int):
         self.decode_head = SegFormerHead(encoder_dims=self.encoder_embed_dims, embed_dim=new_decoder_embed_dim, num_classes=new_num_classes)
 
-    def forward(self, x: torch.Tensor) -> torch.Tensor:
+    def _forward(self, x: torch.Tensor) -> torch.Tensor:
         features = self._backbone(x)
         out = self.decode_head(features)
         out = F.interpolate(out, size=x.shape[2:], mode="bilinear", align_corners=False)
         return out
 
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        if self.is_SWI:
+            return self.sliding_window(x, self.num_classes, self.sliding_window_stride, self.sliding_window_size)
+        else:
+            return self._forward(x)
+
+    def sliding_window(self, img: torch.Tensor, num_classes: int, stride: Tuple[int, int], crop_size: Tuple[int, int]) -> torch.Tensor:
+        """
+        Inference by sliding-window with overlap.
+
+        If h_crop > h_img or w_crop > w_img, the small patch will be used to
+        decode without padding.
+        """
+
+        h_stride, w_stride = stride
+        h_crop, w_crop = crop_size
+        batch_size, _, h_img, w_img = img.size()
+
+        h_grids = max(h_img - h_crop + h_stride - 1, 0) // h_stride + 1
+        w_grids = max(w_img - w_crop + w_stride - 1, 0) // w_stride + 1
+
+        preds = torch.zeros((batch_size, num_classes, h_img, w_img), device=img.device)
+        count_mat = torch.zeros((batch_size, 1, h_img, w_img), device=img.device)
+
+        for h_idx in range(h_grids):
+            for w_idx in range(w_grids):
+                y1 = h_idx * h_stride
+                x1 = w_idx * w_stride
+                y2 = min(y1 + h_crop, h_img)
+                x2 = min(x1 + w_crop, w_img)
+                y1 = max(y2 - h_crop, 0)
+                x1 = max(x2 - w_crop, 0)
+                crop_img = img[:, :, y1:y2, x1:x2]
+                crop_logits = self._forward(crop_img)
+
+                if isinstance(crop_logits, torch.Tensor):
+                    crop_logits = (crop_logits,)
+
+                crop_logits = crop_logits[0]
+
+                preds += F.pad(crop_logits, pad=(int(x1), int(preds.shape[3] - x2), int(y1), int(preds.shape[2] - y2)))
```

Comment:
This increment is somewhat suboptimal. Instead of accumulating `preds[roi] += crop_logits` this code padds the single tile to a full resolution and adds it. This is just ineffective.

Reply:
Fixed on 8a178fa